src/Microsoft.DotNet.ProjectModel/ProjectSystem/ProjectSystemCache.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.ProjectModel;
+using Microsoft.Extensions.ProjectModel.Compilation;
+using Microsoft.Extensions.ProjectModel.Graph;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectModel.ProjectSystem
+{
+    internal class ProjectSystemCache
```

Comment:
I think it! Rather than spreading caching through the app, we have one central "repository" for cachable data.